cargo +nightly install racer Error

I am trying to install *racer*, but I keep getting this error.
I tried setting *nightly* as default, didn't work either.

# Info
<pre>$ rustup show
<b>Default host: </b>x86_64-unknown-linux-gnu

<b>installed toolchains</b>
<b>--------------------</b>

stable-x86_64-unknown-linux-gnu (default)
nightly-x86_64-unknown-linux-gnu

<b>active toolchain</b>
<b>----------------</b>

stable-x86_64-unknown-linux-gnu (default)
rustc 1.37.0 (eae3437df 2019-08-13)
</pre>

<pre>$ rustup update
<b>info: </b>syncing channel updates for &apos;stable-x86_64-unknown-linux-gnu&apos;
<b>info: </b>syncing channel updates for &apos;nightly-x86_64-unknown-linux-gnu&apos;
<b>info: </b>checking for self-updates

   <b>stable-x86_64-unknown-linux-gnu unchanged</b> - rustc 1.37.0 (eae3437df 2019-08-13)
  <b>nightly-x86_64-unknown-linux-gnu unchanged</b> - rustc 1.39.0-nightly (618768492 2019-09-05)
</pre>

# Fail
<pre>$ cargo +nightly install racer
<font color="#55FF55"><b>    Updating</b></font> crates.io index
<font color="#55FF55"><b>  Installing</b></font> racer v2.1.25
<font color="#55FF55"><b>   Compiling</b></font> autocfg v0.1.6
<font color="#55FF55"><b>   Compiling</b></font> libc v0.2.62
<font color="#55FF55"><b>   Compiling</b></font> semver-parser v0.7.0
<font color="#55FF55"><b>   Compiling</b></font> arrayvec v0.4.11
<font color="#55FF55"><b>   Compiling</b></font> rand_core v0.4.2
<font color="#55FF55"><b>   Compiling</b></font> nodrop v0.1.13
<font color="#55FF55"><b>   Compiling</b></font> cfg-if v0.1.9
<font color="#55FF55"><b>   Compiling</b></font> proc-macro2 v0.4.30
<font color="#55FF55"><b>   Compiling</b></font> scopeguard v0.3.3
<font color="#55FF55"><b>   Compiling</b></font> memoffset v0.2.1
<font color="#55FF55"><b>   Compiling</b></font> lazy_static v1.4.0
<font color="#55FF55"><b>   Compiling</b></font> stable_deref_trait v1.1.1
<font color="#55FF55"><b>   Compiling</b></font> rustc-rayon-core v0.2.0
<font color="#55FF55"><b>   Compiling</b></font> log v0.4.8
<font color="#55FF55"><b>   Compiling</b></font> proc-macro2 v1.0.2
<font color="#55FF55"><b>   Compiling</b></font> byteorder v1.3.2
<font color="#55FF55"><b>   Compiling</b></font> unicode-xid v0.1.0
<font color="#55FF55"><b>   Compiling</b></font> smallvec v0.6.10
<font color="#55FF55"><b>   Compiling</b></font> syn v0.15.44
<font color="#55FF55"><b>   Compiling</b></font> rustc-rayon v0.2.0
<font color="#55FF55"><b>   Compiling</b></font> unicode-xid v0.2.0
<font color="#55FF55"><b>   Compiling</b></font> either v1.5.2
<font color="#55FF55"><b>   Compiling</b></font> syn v1.0.5
<font color="#55FF55"><b>   Compiling</b></font> indexmap v1.1.0
<font color="#55FF55"><b>   Compiling</b></font> rustc-ap-graphviz v546.0.0
<font color="#55FF55"><b>   Compiling</b></font> memchr v2.2.1
<font color="#55FF55"><b>   Compiling</b></font> bitflags v1.1.0
<font color="#55FF55"><b>   Compiling</b></font> serde v1.0.99
<font color="#55FF55"><b>   Compiling</b></font> ryu v1.0.0
<font color="#55FF55"><b>   Compiling</b></font> rustc-ap-rustc_target v546.0.0
<font color="#55FF55"><b>   Compiling</b></font> scoped-tls v1.0.0
<font color="#55FF55"><b>   Compiling</b></font> unicode-width v0.1.6
<font color="#55FF55"><b>   Compiling</b></font> quick-error v1.2.2
<font color="#55FF55"><b>   Compiling</b></font> termcolor v1.0.5
<font color="#55FF55"><b>   Compiling</b></font> annotate-snippets v0.6.1
<font color="#55FF55"><b>   Compiling</b></font> itoa v0.4.4
<font color="#55FF55"><b>   Compiling</b></font> rustc-ap-syntax v546.0.0
<font color="#55FF55"><b>   Compiling</b></font> regex-syntax v0.6.12
<font color="#55FF55"><b>   Compiling</b></font> vec_map v0.8.1
<font color="#55FF55"><b>   Compiling</b></font> strsim v0.8.0
<font color="#55FF55"><b>   Compiling</b></font> rustc-ap-rustc_lexer v546.0.0
<font color="#FF5555"><b>error[E0433]</b></font><b>: failed to resolve: could not find `property` in `unicode`</b>
   <font color="#5555FF"><b>--&gt; </b></font>/home/marx/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_lexer-546.0.0/src/lib.rs:672:24
    <font color="#5555FF"><b>|</b></font>
<font color="#5555FF"><b>672</b></font> <font color="#5555FF"><b>| </b></font>        core::unicode::property::Pattern_White_Space(c)
    <font color="#5555FF"><b>| </b></font>                       <font color="#FF5555"><b>^^^^^^^^</b></font> <font color="#FF5555"><b>could not find `property` in `unicode`</b></font>

<font color="#FF5555"><b>error[E0599]</b></font><b>: no method named `is_xid_start` found for type `char` in the current scope</b>
   <font color="#5555FF"><b>--&gt; </b></font>/home/marx/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_lexer-546.0.0/src/lib.rs:689:33
    <font color="#5555FF"><b>|</b></font>
<font color="#5555FF"><b>689</b></font> <font color="#5555FF"><b>| </b></font>            || (c &gt; &apos;\x7f&apos; &amp;&amp; c.is_xid_start())
    <font color="#5555FF"><b>| </b></font>                                <font color="#FF5555"><b>^^^^^^^^^^^^</b></font>

<font color="#FF5555"><b>error[E0599]</b></font><b>: no method named `is_xid_continue` found for type `char` in the current scope</b>
   <font color="#5555FF"><b>--&gt; </b></font>/home/marx/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_lexer-546.0.0/src/lib.rs:708:33
    <font color="#5555FF"><b>|</b></font>
<font color="#5555FF"><b>708</b></font> <font color="#5555FF"><b>| </b></font>            || (c &gt; &apos;\x7f&apos; &amp;&amp; c.is_xid_continue())
    <font color="#5555FF"><b>| </b></font>                                <font color="#FF5555"><b>^^^^^^^^^^^^^^^</b></font>

<font color="#FF5555"><b>error</b></font><b>: aborting due to 3 previous errors</b>

<b>Some errors have detailed explanations: E0433, E0599.</b>
<b>For more information about an error, try `rustc --explain E0433`.</b>
<font color="#FF5555"><b>error</b></font><b>:</b> Could not compile `rustc-ap-rustc_lexer`.
<font color="#FFFF55"><b>warning</b></font><b>:</b> build failed, waiting for other jobs to finish...
<font color="#FF5555"><b>error</b></font><b>:</b> failed to compile `racer v2.1.25`, intermediate artifacts can be found at `/tmp/cargo-installsc6zOC`

Caused by:
  build failed
</pre>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,7 +24,7 @@ debug = false # because of #1005
 [dependencies]
 bitflags = "1.0"
 log = "0.4"
-rustc-ap-syntax = "546.0.0"
+rustc-ap-syntax = "581.0.0"
 env_logger = "0.6"
 clap = "2.32"
 lazy_static = "1.2"

--- a/src/racer/ast.rs
+++ b/src/racer/ast.rs
@@ -315,7 +315,7 @@ fn destructure_pattern_to_ty(
             for child in children {
                 if point_is_in_span(point, &child.span) {
                     return typeinf::get_struct_field_type(
-                        &child.node.ident.name.as_str(),
+                        &child.ident.name.as_str(),
                         &m,
                         session,
                     )
@@ -327,7 +327,7 @@ fn destructure_pattern_to_ty(
                         }
                     })
                     .and_then(|ty| {
-                        destructure_pattern_to_ty(&child.node.pat, point, &ty, scope, session)
+                        destructure_pattern_to_ty(&child.pat, point, &ty, scope, session)
                     });
                 }
             }
@@ -745,7 +745,7 @@ impl<'c, 's, 'ast> visit::Visitor<'ast> for ExprTypeVisitor<'c, 's> {
                 }
             }
             ExprKind::Mac(ref m) => {
-                if let Some(name) = m.node.path.segments.last().map(|seg| seg.ident) {
+                if let Some(name) = m.path.segments.last().map(|seg| seg.ident) {
                     // use some ad-hoc rules
                     if name.as_str() == "vec" {
                         let path = RacerPath::from_iter(
@@ -868,10 +868,6 @@ impl<'ast> visit::Visitor<'ast> for StructVisitor {
     fn visit_variant_data(
         &mut self,
         struct_definition: &ast::VariantData,
-        _: ast::Ident,
-        _: &ast::Generics,
-        _: ast::NodeId,
-        _: Span,
     ) {
         for field in struct_definition.fields() {
             let ty = Ty::from_ast(&field.ty, &self.scope);
@@ -894,7 +890,7 @@ pub struct TypeVisitor<'s> {
 
 impl<'ast, 's> visit::Visitor<'ast> for TypeVisitor<'s> {
     fn visit_item(&mut self, item: &ast::Item) {
-        if let ItemKind::Ty(ref ty, _) = item.node {
+        if let ItemKind::TyAlias(ref ty, _) = item.node {
             self.name = Some(item.ident.name.to_string());
             self.type_ = Ty::from_ast(&ty, self.scope);
             debug!("typevisitor type is {:?}", self.type_);
@@ -1001,7 +997,7 @@ impl<'ast> visit::Visitor<'ast> for EnumVisitor {
             for variant in &enum_definition.variants {
                 let source_map::BytePos(point) = variant.span.lo();
                 self.values
-                    .push((variant.node.ident.to_string(), point.into()));
+                    .push((variant.ident.to_string(), point.into()));
             }
         }
     }

--- a/src/racer/ast_types.rs
+++ b/src/racer/ast_types.rs
@@ -271,6 +271,7 @@ pub enum Pat {
     Slice,
     Mac,
     Rest,
+    Or,
 }
 
 impl Pat {
@@ -304,7 +305,7 @@ impl Pat {
                 let path = Path::from_ast(path, scope);
                 let fields = fields
                     .iter()
-                    .map(|fld| FieldPat::from_ast(&fld.node, scope))
+                    .map(|fld| FieldPat::from_ast(&fld, scope))
                     .collect();
                 Pat::Struct(path, fields)
             }
@@ -333,6 +334,7 @@ impl Pat {
             PatKind::Paren(pat) => Pat::from_ast(&pat.node, scope),
             PatKind::Mac(_) => Pat::Mac,
             PatKind::Rest => Pat::Rest,
+            PatKind::Or(_) => Pat::Or,
         }
     }
 }

